### PR TITLE
[TIMOB-24051] Add the ability to remove all Ti.App.Properties at once

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-24051](https://jira.appcelerator.org/browse/TIMOB-24051)

Implement `Ti.App.Properties.removeAllProperties()`.

```javascript
Ti.App.Properties.setString('test_removeAllProperties1', 'test1');
Ti.App.Properties.setString('test_removeAllProperties2', 'test2');
Ti.App.Properties.removeAllProperties();
should(Ti.App.Properties.hasProperty("test_removeAllProperties1")).be.false;
should(Ti.App.Properties.hasProperty("test_removeAllProperties2")).be.false;
```

### CODE REVIEW

**Comment by garymathews** [comment]
test this please

File: Examples/NMocha/src/Assets/ti.app.properties.test.js [hunk 1]
```diff
@@ -237,4 +237,12 @@ describe('Titanium.App.Properties', function () {
 		finish();
 	});
 
+	it('removeAllProperties should remove all properties', function (finish) {
+		Ti.App.Properties.setString('test_removeAllProperties1', 'test1');
+		Ti.App.Properties.setString('test_removeAllProperties2', 'test2');
+		Ti.App.Properties.removeAllProperties();
```
Comment (garymathews) [inline Examples/NMocha/src/Assets/ti.app.properties.test.js]: Check the properties exist before removal

**Comment by garymathews** [comment]
test this please

**Comment by infosia** [comment]
All check passed (finally), merged.

**Pull request merged**